online-dashboard-api/app/Http/Controllers/AuthenticationController.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\LoginRequest;
+use App\Http\Requests\RefreshRequest;
+use App\Http\Requests\SignupRequest;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Session;
+use Laravel\Passport\Http\Controllers\AccessTokenController;
+use Laravel\Passport\RefreshTokenRepository;
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Throwable;
+
+class AuthenticationController extends Controller
+{
+    public function signUp(SignupRequest $request)
+    {
+
+        try {
+            //creating a new user
+            $user = User::create([
+                'name' => $request->name,
+                'email' => $request->email,
+                'password' => $request->password,
+            ]);
+
+            //generate access token using helper function
+            $tokenData = generateAccessToken($user, $request->password);
+
+            info($user);
+
+            info($tokenData);
+
+            //check if token generation failed
+            if (! $tokenData) {
+                //delete the user if token generation fails
+                $user->delete();
+
+                return response()->json([
```

Comment:
Create a seperate HTTP file ApiResponse and pass the resonse to the function to render data!

Reply:
done